test/functional/test_framework/test_node.py

```diff
@@ -232,8 +232,18 @@ def wait_for_rpc_connection(self):
         poll_per_s = 4
         for _ in range(poll_per_s * self.rpc_timeout):
             if self.process.poll() is not None:
+                # Attach abrupt shutdown error/s to the exception message
+                str_error = ''
+                self.stderr.seek(0)
+                err = ''.join(line for line in self.stderr.read().decode('utf-8').strip() if line[0] != "*")
```

Comment:
This list comprehension results in a list of characters rather than a list of lines, and filters out _all_ `*` characters, rather than only lines starting with `*`. I assume what you intended would look more like this:
```suggestion
                err = ''.join(line.decode('utf-8') for line in self.stderr if not line.startswith(b"*"))
```
(On the other hand, do we need this `*` filter at all?)

I tested with the following patch:
```diff
diff --git a/src/init.cpp b/src/init.cpp
index 33389e695..e6509cc85 100644
--- a/src/init.cpp
+++ b/src/init.cpp
@@ -1661,6 +1661,7 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
     }
 
     chainman.m_thread_load = std::thread(&util::TraceThread, "initload", [=, &chainman, &args, &node] {
+        throw std::logic_error("I'm an ugly error.\n*** This line should be skipped.\nThis line should be visible. ***");
         // Import blocks
         ImportBlocks(chainman, vImportFiles);
         if (args.GetBoolArg("-stopafterblockimport", DEFAULT_STOPAFTERBLOCKIMPORT)) {
```

Reply:
> On the other hand, do we need this * filter at all?

We need it to present the crashed node's identification within the error message. Because the stderr message contain the * separator ([link](https://github.com/bitcoin/bitcoin/blob/5b4478418b24422885ae4d5c9a41a45455a72618/src/util/exception.cpp#L40)).

In other words, right now the format is:

```
****************
<node_datadir> abruptly aborted with error:

<error message>
****************
```

And what you propose would looks like:

```
<node_datadir> abruptly aborted with error:
****************
<error message>
****************
```

Which I think that is harder to read.

Reply:
Okay I see, that makes sense.